pkg/utils/helper.go

```diff
@@ -90,28 +106,59 @@ func GetHostName() (string, error) {
 	return strings.TrimSuffix(string(hostname), "\n"), nil
 }
 
-// GetAPIServerFQDN gets the API Server FQDN from the kubeconfig file
-func GetAPIServerFQDN() (string, error) {
-	output, err := RunCommandOnHost("cat", "/var/lib/kubelet/kubeconfig")
-
-	if err != nil {
-		return "", fmt.Errorf("Can't open kubeconfig file: %+v", err)
-	}
-
+//ParseAPIServerFQDNFromKubeConfig parses a kubeConfig and returns the APIServerFQDN
+func ParseAPIServerFQDNFromKubeConfig(output string) (string, error) {
 	lines := strings.Split(output, "\n")
 	for _, line := range lines {
 		index := strings.Index(line, "server: ")
 		if index >= 0 {
 			fqdn := line[index+len("server: "):]
-			fqdn = strings.Replace(fqdn, "https://", "", -1)
-			fqdn = strings.Replace(fqdn, ":443", "", -1)
-			return fqdn, nil
+			fqdnurl, err := url.Parse(fqdn)
+			if err != nil {
+				return "", fmt.Errorf("Fail to parse url from fqdn: %s", fmt.Sprint(err)+": "+fqdn)
+			}
+
+			host, _, err := net.SplitHostPort(fqdnurl.Host)
+			if err != nil {
+				return "", fmt.Errorf("Fail to split host port from fqdnurl: %s", fmt.Sprint(err)+": "+fqdnurl.String())
+			}
+
+			return host, nil
 		}
 	}
-
 	return "", errors.New("Could not find server definitions in kubeconfig")
 }
 
+//ReadKubeletConfig reads the kubeletConfig from the node
+func ReadKubeletConfig() (string, error) {
+	if IsKubernetesInDocker() {
+		output, err := RunCommandOnHost("cat", "/etc/kubernetes/kubelet.conf")

```

Comment:
I think we can avoid runCmd here

Reply:
As above, how can we do this via nsenter?

Reply:
Maybe this? https://pkg.go.dev/github.com/kata-containers/runtime/virtcontainers/pkg/nsenter?utm_source=godoc. Regardless this should go in a separate PR

Reply:
Hmm might not be a go https://www.weave.works/blog/linux-namespaces-and-go-don-t-mix